mbf_abstract_nav/src/move_base_action.cpp

```diff
@@ -292,27 +291,22 @@ void MoveBaseAction::actionGetPathDone(
       else
       {
         // copy result from get_path action
-        move_base_result.outcome = result.outcome;
-        move_base_result.message = result.message;
-        move_base_result.dist_to_goal = static_cast<float>(mbf_utility::distance(robot_pose_, goal_pose_));
-        move_base_result.angle_to_goal = static_cast<float>(mbf_utility::angle(robot_pose_, goal_pose_));
-        move_base_result.final_pose = robot_pose_;
-
-        ROS_WARN_STREAM_NAMED("move_base", "Abort the execution of the planner: " << result.message);
+        fillMoveBaseResult(get_path_result, move_base_result);
+        ROS_WARN_STREAM_NAMED("move_base", "Abort the execution of the planner: " << get_path_result.message);
         goal_handle_.setAborted(move_base_result, state.getText());
       }
+      action_state_ = FAILED;
       break;
 
     case actionlib::SimpleClientGoalState::PREEMPTED:
-      // the get_path action has been preempted.
-
-      // copy result from get_path action
-      move_base_result.outcome = result.outcome;
-      move_base_result.message = result.message;
-      move_base_result.dist_to_goal = static_cast<float>(mbf_utility::distance(robot_pose_, goal_pose_));
-      move_base_result.angle_to_goal = static_cast<float>(mbf_utility::angle(robot_pose_, goal_pose_));
-      move_base_result.final_pose = robot_pose_;
-      goal_handle_.setCanceled(move_base_result, state.getText());
+      ROS_INFO_STREAM_NAMED("move_base", "The action \"" << "get_path" << "\" has been preempted!");
+      if (action_state_ == CANCELED)
```

Comment:
move base has been preempted. otherwise we have preempted internally, (what I don't think can happen) so we don't end the action